src/diagram/components/dialog/edit-variable-dialog.tsx

```diff
@@ -1,74 +1,127 @@
 import React, { useMemo } from "react";
 import { observer } from "mobx-react-lite";
-import { applySnapshot, getSnapshot } from "mobx-state-tree";
-
+import { applySnapshot, getSnapshot, IAnyComplexType, Instance } from "mobx-state-tree";
 import { NumberRow } from "./number-row";
 import { TextRow } from "./text-row";
 import { TextAreaRow } from "./text-area-row";
 import { Variable, VariableType } from "../../models/variable";
 import { kMaxNameCharacters, kMaxNotesCharacters, processName } from "../../utils/validate";
+import classNames from "classnames";
+import { IconInputCard } from "./icon-input-card";
+import { IconOutputCard } from "./icon-output-card";
+import { ErrorMessage } from "../error-message";
 
 import "./dialog.scss";
 
-interface IEditVariableDialogContent {
-  variable: VariableType;
+export interface IEditVariableDialogContent {
+  variable?: VariableType;
+  variableClone: VariableType;
 }
-export const EditVariableDialogContent = observer(({ variable }: IEditVariableDialogContent) => {
+
+export const EditVariableDialogContent = observer(({ variable, variableClone }: IEditVariableDialogContent) => {
+  // Because variableClone was created from a snapshot, we will not be able to access 
+  // computed values related to the original variable's inputs. We need to get those values
+  // from the original variable.
```

Comment:
I think it would be worth explaining why we have a variable clone in the first place: so the user can modify its properties directly, but those changes won't be saved unless the ok button is pushed.

Reply:
Good point. I've updated the comment.